apluslms_roman/backends/docker.py

```diff
@@ -111,7 +111,7 @@ def build(self, task, observer):
                 return BuildResult(-1, error, step)
             except KeyboardInterrupt:
                 observer.step_cancelled(step)
-                raise
+                return BuildResult(step=step)
```

Comment:
I think we should include the cancellation info in the buildresult object.

Thus, I think new BuldResult would include following:

| "state" | .ok | .cancelled | .failed | .code |
| ---------- | ---------- | ------- | -------- | --- |
| SUCCEEDED | True | False | Fasle | 0 |
| CANCELLED | False | True | False | 0 |
| FAILED | Fase | False | True | != 0 |

`.ok` needs to be changed from property to a variable and these could be properties `.cancelled = not .ok and .code == 0` and `.failed = not .ok and .code != 0`.

After that, we should change `Observer.result_msg` to include `_state_update(step, state)` based on the value in `result` object. This would trigger step state update, if it was not done yet (should have been done, but in theory). This ensure that the state send wiht `RESULT_MSG` is always correct.

Reply:
Is cancelled a necessary property? If the step neither succeeded or failed we know it was cancelled.